src/ir/objc.rs

```diff
@@ -194,6 +207,41 @@ impl ObjCInterface {
             }
             CXChildVisit_Continue
         });
+
+        if interface.is_category() {
+            // If this interface is a category, we need to find the interface that this category
+            // extends.
+            let needle = format!("{}", interface.name());
+            debug!(
+                "Category {} belongs to {}, find the item",
+                interface.rust_name(),
+                needle
+            );
+
+            // This is pretty gross but using the ItemResolver doesn't yield a mutable reference.
+            if let Some(real_interface_id) = real_interface_id_for_category {
+                if let Some(real_interface_item) = ctx.get_item_mut(real_interface_id) {
+                    if let Some(ty) = real_interface_item.kind().as_type() {
+                        if let TypeKind::ResolvedTypeRef(item_id) = ty.kind() {
+                            let real_interface_id: ItemId = item_id.into();
+                            let ty = ctx.get_item_mut(real_interface_id).unwrap().kind_mut().as_type_mut().unwrap();
```

Comment:
I don't like this line but am not sure what to do. Would adding a new function that's like...
```
fn get_parent_ty(ctx: BindgenContext, parent_id: ItemId) -> Option<&mut Type>
```
be reasonable? The benefit would be that we could use `?` on all the options.

Reply:
I did this.